program/src/processor.rs

```diff
@@ -22,22 +21,26 @@ use {
         sysvar::{epoch_rewards::EpochRewards, stake_history::StakeHistorySysvar, Sysvar},
         vote::{program as solana_vote_program, state::VoteState},
     },
-    std::collections::HashSet,
+    std::{collections::HashSet, mem::MaybeUninit},
 };
 
 // TODO a nice change would be to pop an account off the queue and discard if
 // its a gettable sysvar ie, allow people to omit them from the accounts list
 // without breaking compat to be done after release, we keep the existing
 // interface for all instructions for compat with firedancer
 
+// TODO undecided if we do it in initial release or in a future release...
+// but we could use StakeError much more extensively and impl PrintProgramError
+
 fn get_vote_state(vote_account_info: &AccountInfo) -> Result<Box<VoteState>, ProgramError> {
     if *vote_account_info.owner != solana_vote_program::id() {
         return Err(ProgramError::IncorrectProgramId);
     }
 
-    let mut vote_state = Box::<VoteState>::default();
-    VoteState::deserialize_into(&vote_account_info.data.borrow(), &mut vote_state)
+    let mut vote_state = Box::new(MaybeUninit::uninit());
+    VoteState::deserialize_into_uninit(&vote_account_info.try_borrow_data()?, vote_state.as_mut())
         .map_err(|_| ProgramError::InvalidAccountData)?;
+    let vote_state = unsafe { Box::from_raw(Box::into_raw(vote_state) as *mut VoteState) };
```

Comment:
Considering your code is essentially the exact implementation of `Box<MaybeUninit<T>>::assume_init()` at https://doc.rust-lang.org/src/alloc/boxed.rs.html#937, this looks correct to me!

More seriously, I had to do the same thing of replicating a function that wasn't available yet on MaybeUninit for slices, so this approach is valid.

The one thing I find weird is calling `as_mut()` to convert from `Box<MaybeUninit<T>>` to `&mut MaybeUninit<T>` when calling `deserialize_into_uninit`, only for that function to call `as_mut_ptr()` for the private `deserialize_into_ptr` function, when it would be a bit more ergonomic to call directly into `deserialize_into_ptr`, but that's not a problem with this code. And I imagine that the compiler just optimizes that away.